### PR TITLE
Router & Role check upgrade [issue #443]

### PR DESCRIPTION
**Pull request by rajatsandeepsen**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # ( #443)

## Type of change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] HotFix (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by rajatsandeepsen** [comment]
Screenshot 2023-07-16 114204

File: src/services/authCheck.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+interface AuthRoutesProps {
+    Children: JSX.Element
+    roles: Role[]
+}
+
+let localRoles = [] as Role[]
+
+export const refreshRoles = () => {
+     localRoles = fetchLocalStorage<UserInfo>('userInfo')?.roles || []
+     console.log("refreshRoles", localRoles)
```
Comment (rajatsandeepsen) [inline src/services/authCheck.tsx:15]: remove console.log( )

File: src/App.tsx [hunk 2]
```diff
@@ -52,6 +52,13 @@ const HackathonRegistration = lazy(() => import("./modules/Dashboard/modules/Hac
 const LandingPage = lazy(() => import("./modules/Public/LearningCircles/modules/LandingPage/LandingPage"));
 const ConnectDiscord = lazy(() => import("./modules/Dashboard/modules/ConnectDiscord/pages/ConnectDiscord"));
 
+
+import { roles } from "./services/types";
+import SecureAuthRoutes from "./services/authCheck";
```
Comment (rajatsandeepsen) [inline src/App.tsx:60]: convert to @/MulearnServices

File: src/modules/Common/Authentication/services/apis.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ import {
     publicGateway
 } from "../../../../services/apiGateways";
 import { authRoutes, dashboardRoutes } from "../../../../services/urls";
+import { refreshRoles } from "@/MuLearnServices/authCheck";
```
Comment (rajatsandeepsen) [inline src/modules/Common/Authentication/services/apis.ts:9]: convert import to @/MuLearnServices

**Review by AswinAsok — APPROVED** [review]

**Review by AswinAsok — APPROVED** [review]

**Pull request merged**